arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,903 @@
+                        "name": "associationName",
+                        "description": "The name of the commitment association to move.",
+                        "in": "path",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "moveRequest",
+                        "description": "Specifies the destination commitment plan.",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/MoveCommitmentAssociationRequest"
+                        }
+                    },
+                    {
+                        "$ref": "#/parameters/APIVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/CommitmentAssociation"
+                        }
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{planName}": {
+            "get": {
+                "tags": [
+                    "CommitmentPlan"
+                ],
+                "operationId": "CommitmentPlan_GetCommitmentPlan",
+                "description": "Gets a commitment plan.",
+                "parameters": [
+                    {
+                      "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "name": "resourceGroupName",
+                        "description": "The resource group the resource is under.",
+                        "in": "path",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "planName",
+                        "description": "The name of the commitment plan resource.",
+                        "in": "path",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "$ref": "#/parameters/APIVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/CommitmentPlan"
+                        }
+                    }
+                }
+            },
+            "put": {
+                "tags": [
+                    "CommitmentPlan"
+                ],
+                "operationId": "CommitmentPlan_PutCommitmentPlan",
+                "description": "Creates or updates a commitment plan resource.",
```

Comment:
The service supports CommitmentPlan_PutCommitmentPlan, CommitmentPlan_PostCommitmentPlan, CommitmentPlan_PatchCommitmentPlan. What is the difference between them apart from the REST request type. Do you really need all of them?